plugins/janus_streaming.c

```diff
@@ -6433,7 +6433,11 @@ static int janus_streaming_rtsp_connect_to_server(janus_streaming_mountpoint *mp
 	int asport = 0, asport_rtcp = 0;
 	multiple_fds audio_fds = {-1, -1};
 
-	janus_mutex_lock(&mountpoints_mutex);
+	if(g_atomic_int_get(&mp->destroyed))
+       return -8;
+	janus_mutex_lock(&mp_mutex);
+	if(g_atomic_int_get(&mp->destroyed))
+	   return -8;
```

Comment:
Indentation is broken. Please use tabs, not spaces, as that's the code style. Besides, you're locking and returning without an unlock after the second if (which checks the same thing again? why?)